proof-producer/bin/proof-producer/include/nil/proof-generator/assignment_table_writer.hpp

```diff
@@ -0,0 +1,162 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2024 Daniil Kogtev <oclaw@nil.foundation>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//---------------------------------------------------------------------------//
+
+#ifndef PROOF_GENERATOR_ASSIGNMENT_TABLE_WRITER_HPP
+#define PROOF_GENERATOR_ASSIGNMENT_TABLE_WRITER_HPP
+
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/assert.hpp>
+#include <ostream>
+
+#include <nil/blueprint/blueprint/plonk/assignment.hpp>
+#include <nil/crypto3/marshalling/algebra/types/field_element.hpp>
+#include <nil/crypto3/zk/snark/arithmetization/plonk/assignment.hpp>
+#include <nil/marshalling/types/integral.hpp>
+
+
+namespace nil {
+    namespace proof_generator {
+
+        template <typename Endianness, typename BlueprintField>
+        class assignment_table_writer {
+            public:                
+                using Column = nil::crypto3::zk::snark::plonk_column<BlueprintField>;
+                using ArithmetizationType = nil::crypto3::zk::snark::plonk_constraint_system<BlueprintField>;
+
+                using AssignmentTableType = nil::blueprint::assignment<ArithmetizationType>;
```

Comment:
useless

Reply:
removed, it was used only to shorten another template description